Tools/TransformTool/TransformTool.cs

```diff
@@ -537,12 +539,22 @@ public void DropHeldObjects(Node node)
 			this.ClearSnappingState(rayOrigin);
 		}
 
-		void Translate(Vector3 delta, Transform rayOrigin, bool constrained)
+		void Translate(Vector3 delta, Transform rayOrigin, ConstrainedAxis constraints)
 		{
-			if (constrained)
-				m_TargetPosition += delta;
-			else
-				this.ManipulatorSnap(rayOrigin, Selection.transforms, ref m_TargetPosition, ref m_TargetRotation, delta);
+			switch (constraints)
+			{
+				case ConstrainedAxis.X | ConstrainedAxis.Y:
+				case ConstrainedAxis.Y | ConstrainedAxis.Z:
+				case ConstrainedAxis.X | ConstrainedAxis.Z:
```

Comment:
Should the axis constraints be adjusting the delta here to knock out the constrained/non-constrained axes?